athena-mysql/src/test/java/com/amazonaws/athena/connectors/mysql/MySqlQueryStringBuilderTest.java

```diff
@@ -0,0 +1,110 @@
+import com.amazonaws.athena.connector.lambda.domain.predicate.SortedRangeSet;
+import com.amazonaws.athena.connector.lambda.metadata.GetSplitsRequest;
+import com.amazonaws.athena.connector.lambda.metadata.GetSplitsResponse;
+import com.amazonaws.athena.connector.lambda.metadata.GetTableLayoutRequest;
+import com.amazonaws.athena.connectors.jdbc.connection.DatabaseConnectionConfig;
+import com.google.common.collect.ImmutableMap;
+import org.apache.arrow.vector.types.pojo.Schema;
+import org.apache.logging.log4j.core.config.plugins.validation.Constraint;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+import org.mockito.Mockito;
+
+import static org.junit.Assert.assertEquals;
+
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.SQLException;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Map;
+
+import static com.amazonaws.athena.connector.lambda.domain.predicate.Constraints.DEFAULT_NO_LIMIT;
+import com.amazonaws.athena.connector.lambda.domain.spill.S3SpillLocation;
+import com.amazonaws.athena.connector.lambda.domain.predicate.ValueSet;
+import com.amazonaws.athena.connectors.jdbc.manager.DefaultJdbcFederationExpressionParser;
+
+public class MySqlQueryStringBuilderTest
+{
+    private String catalogName = "testCatalog";
+    private String schemaName = "testSchema";
+    private String tableName = "testTable";
+    private BlockAllocator allocator = new BlockAllocatorImpl("test-allocator-id");
+    private Connection connection = Mockito.mock(Connection.class, Mockito.RETURNS_DEEP_STUBS);
+    private S3SpillLocation s3SpillLocation = S3SpillLocation.newBuilder().withIsDirectory(true).build();
+    Split.Builder splitBuilder = Split.newBuilder(s3SpillLocation, null).add("partition_name", String.valueOf("p0"));
+    private MySqlQueryStringBuilder mySqlQueryStringBuilder = new MySqlQueryStringBuilder("`", new DefaultJdbcFederationExpressionParser());
+    Schema schema = SchemaBuilder.newBuilder()
+        .addField(FieldBuilder.newBuilder("testCol1", org.apache.arrow.vector.types.Types.MinorType.VARCHAR.getType()).build())
+        .addField(FieldBuilder.newBuilder("testCol2", org.apache.arrow.vector.types.Types.MinorType.INT.getType()).build())
+        .addField(FieldBuilder.newBuilder("testCol3", org.apache.arrow.vector.types.Types.MinorType.VARCHAR.getType()).build())
+        .addField(FieldBuilder.newBuilder("testCol4", org.apache.arrow.vector.types.Types.MinorType.FLOAT8.getType()).build())
+        .addField(FieldBuilder.newBuilder("partition_name", org.apache.arrow.vector.types.Types.MinorType.VARCHAR.getType()).build())
+        .build();
+
+    @Test
+    public void generateSqlIsNotNull() throws Exception
+    {
+        Map<String, ValueSet> constraintsMap = new HashMap<>();
+        constraintsMap.put("testCol2", SortedRangeSet.of(false, Range.all(allocator, org.apache.arrow.vector.types.Types.MinorType.INT.getType())));
```

Comment:
nit, you can init and add the map values in one line using ImmutableMap.of(key, value, key2, value2, ...).

It's generally cleaner to avoid initing a collection and later adding or modifying it (called the init-then-modify antipattern). This here isn't maybe only a violation of it from a purists perspective, but you can imagine how it can lead to cleaner code. so this comment applies to both tests in this file